controllers/machine_controller_phases.go

```diff
@@ -293,15 +293,15 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, cluster
 
 	// Get and set Status.Addresses from the infrastructure provider.
 	err = util.UnstructuredUnmarshalField(infraConfig, &m.Status.Addresses, "status", "addresses")
-	if err != nil && err != util.ErrUnstructuredFieldNotFound {
+	if err != nil && err != capierrors.ClusterFieldNotFound("field not found") {
```

Comment:
why is this called ClusterFieldNotFound while fetching a Machine field?
Why is it changing the current implementation?

Reply:
This is the name I had given to the function which returns a UnStructuredError error type , I can change the name if needed

Reply:
@DiptoChakrabarty This should be changed to capierrors.UnstructuredFieldNotFound